hathor/indexes/deps_index.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Iterator, List
+
+from hathor.transaction import BaseTransaction
+from hathor.transaction.transaction_metadata import ValidationState
+
+if TYPE_CHECKING:  # pragma: no cover
+    from hathor.transaction.storage import TransactionStorage
+
+
+INF_HEIGHT: int = 2**32 - 1
+
+
+def get_validation_state(tx_storage: 'TransactionStorage', tx: bytes) -> ValidationState:
+    """Query database for the validation state of a transaction, returns INITIAL when tx does not exist."""
+    tx_meta = tx_storage.get_metadata(tx)
+    if tx_meta is None:
+        return ValidationState.INITIAL
+    return tx_meta.validation
+
+
+class DepsIndex(ABC):
```

Comment:
Add docstring explaining what kind of queries are accelerated by this index and where it's used.

Reply:
That would be great. Will do.

Reply:
Thanks!